docs/testplan/smart-switch/high-availability/keysight_ha_testplan.md

```diff
@@ -0,0 +1,259 @@
+Each test case will use the following metrics to measure performance of the HA
+failover/switchover for all test cases.
+
+The following metrics shall be collected Before and After switchover is initiated.
+
+| Performance Metric                  | Achieved |
+|:------------------------------------|:--------:|
+| Concurrent Connections              |         |
+| CPS                                 |         |
+| PPS                                 |         |
+| TCP Client Failures Resets Sent     |         |
+| TCP Client Failures Resets Recieved |         |
+| TCP Client Failures Resets Retries  |         |
+| TCP Server Failures Resets Sent     |         |
+| TCP Server Failures Resets Recieved |         |
+| TCP Server Failures Resets Retries  |         |
+
+As soon as failover/switchover is initiated a timer will be initiated to measure DPU pair switchover.
+
+| Failover/Switchover<br/>Type |                   Event                    | Convergence (s) |
+|:------------------------|:------------------------------------------:| :---: |
+|                         | Switchover time from <br/>Active-to-Standby DPU | 0  |
+
+### Test case1 CPS Without HA Enabled
+#### Test Objective 1
+* Baseline performance of the SmartSwitch system before HA sets are established.
+* Test metrics will be collected on a single DPU within a SmartSwitch
+
+![HA CPS Without HA Enabled](images/ha_test_cps_ha_not_enabled.svg)
+#### Steps for Test Case 1
+* Configure 1 DPUs networked within a SmartSwitch with HA not enable.
+* The SmartSwitch configuration will consist of 1 DPU sharing the same network configurations
+such as: IPs, MACs, VLan/VxLan, ENIs.
+* The UHD will be configured to split traffic into two separate entities.
+* There will be a physical link between front panel ports of Tor1 both DPU0 and DPU1.
+* The HA set shall be configured within the same SmartSwitch.
+* Verify links are up and start all protocols and verify traffic is established.
+* Enable csv logging and check the state of the DPUs through the API.
+* Apply and start traffic stateful and stateless.
+* Verify traffic is flowing without any loss.
+* Through mgmt port poweroff or reboot Tor1.
+* Mark start time at beginning of test as DPU is removed from topology.
+* There should be 100% failure with the Active DPU during switchover.
+* DPU1 shall become the new Active DPU in the HA set.
+* Mark time when the Standby DPU becomes Active and fully running traffic.
+* Measure convergence time from start of the link removal on DPU0 and DPU1 switchover.
+* Using traffic generator tools to verify metrics associated with number of Concurrent
+Connections, Connection rate, and TCP/UDP failures collect data before, during and after
+switchover.
+
+
+### Test case2 Perfect Sync Between HA Sets
+#### Test Objective 2
+
+![HA Perfect Sync Between HA Sets](images/ha_test_perfect_sync.svg)
+#### Steps for Test Case 2
+* Configure 2 DPUs networked across two SmartSwitches forming an HA set.
```

Comment:
These steps are the same for every test case, so it will be better to extract them out and reference them as common steps.

Reply:
Also many things are shared in these cases below. It might be much easier for people to get the important part by using tables with columns like: action to apply and expectations.

Reply:
Feel free to refer to the HA test plan table format.